include/swift/AST/DiagnosticEngine.h

```diff
@@ -309,8 +309,22 @@ namespace swift {
         : OpeningQuotationMark("'"), ClosingQuotationMark("'"),
           AKAFormatString("'%s' (aka '%s')"),
           OpaqueResultFormatString("'%s' (%s of '%s')") {}
+
+    /// When formatting fix-it arguments, don't include quotes or other
+    /// additions which would result in invalid code.
+    static DiagnosticFormatOptions formatForFixIts() {
+      return DiagnosticFormatOptions("", "", "%s", "%s");
```

Comment:
It would be better if we had a way to turn off AKA formatting completely, since it will do unnecessary work. But we still need the quotation bits, and the OpaqueResultFormatString will never show up in practice, so this is okay.

Reply:
I think if/when more differences between fix-it and regular formatting arise we can think about adding a flag to DiagnosticFormatOptions to control this.